src/debugger/logTracker.ts

```diff
@@ -0,0 +1,84 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2023 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from "vscode";
+
+/**
+ * Factory class for building LoggingDebugAdapterTracker
+ */
+export class LoggingDebugAdapterTrackerFactory implements vscode.DebugAdapterTrackerFactory {
+    createDebugAdapterTracker(
+        session: vscode.DebugSession
+    ): vscode.ProviderResult<vscode.DebugAdapterTracker> {
+        return new LoggingDebugAdapterTracker(session.id);
+    }
+}
+
+interface iOutputEventBody {
+    category: string;
+    output: string;
+}
+
+interface iDebugMessage {
```

Comment:
Is the leading I for interface here intentionally lowercase?

Reply:
Never been really sure how to name interfaces in JS. I'm not a great fan of the `i` prefix but have seen it quite a bit. I might just get rid of the prefix.

Reply:
That is common on Windows, but it tends to use `I` instead of `i`.